tests/benchmark/test_baseline_benchmarks.py

```diff
@@ -0,0 +1,322 @@
+    impact everything else """
+
+    empty_filepath = _empty_filepath()
+    empty_file_info = (
+        "name-emptyfile-file",
+        _empty_filepath(),
+        "modname-emptyfile-mod",
+    )
+    lot_of_files = 500
+
+    def test_baseline_benchmark_j1(self, benchmark):
+        """ Establish a baseline of pylint performance with no work
+
+        We will add extra Checkers in other benchmarks.
+
+        Because this is so simple, if this regresses something very serious has happened
+        """
+        linter = PyLinter(reporter=Reporter())
+        fileinfos = [self.empty_filepath]  # Single file to end-to-end the system
+        assert linter.config.jobs == 1
+        assert len(linter._checkers) == 1, "Should just have 'master'"
+        benchmark(linter.check, fileinfos)
+        assert linter.msg_status == 0, (
+            "Expected no errors to be thrown: %s"
+            % pprint.pformat(linter.reporter.messages)
+        )
+
+    def test_baseline_benchmark_j10(self, benchmark):
+        """ Establish a baseline of pylint performance with no work across threads
+
+        Same as `test_baseline_benchmark_j1` but we use -j10 with 10 fake files to
+        ensure end-to-end-system invoked.
+
+        Because this is also so simple, if this regresses something very serious has
+        happened.
+        """
+        linter = PyLinter(reporter=Reporter())
+        linter.config.jobs = 10
+
+        # Create file per worker, using all workers
+        fileinfos = [self.empty_filepath for _ in range(linter.config.jobs)]
+
+        assert linter.config.jobs == 10
+        assert len(linter._checkers) == 1, "Should have 'master'"
+        benchmark(linter.check, fileinfos)
+        assert linter.msg_status == 0, (
+            "Expected no errors to be thrown: %s"
+            % pprint.pformat(linter.reporter.messages)
+        )
+
+    def test_baseline_benchmark_check_parallel_j10(self, benchmark):
+        """ Should demonstrate times very close to `test_baseline_benchmark_j10` """
+        linter = PyLinter(reporter=Reporter())
+
+        # Create file per worker, using all workers
+        fileinfos = [self.empty_file_info for _ in range(linter.config.jobs)]
+
+        assert len(linter._checkers) == 1, "Should have 'master'"
+        benchmark(check_parallel, linter, jobs=10, files=fileinfos)
+        assert linter.msg_status == 0, (
+            "Expected no errors to be thrown: %s"
+            % pprint.pformat(linter.reporter.messages)
+        )
+
+    def test_baseline_lots_of_files_j1(self, benchmark):
+        """ Establish a baseline with only 'master' checker being run in -j1
+
+        We do not register any checkers except the default 'master', so the cost is just
+        that of the system with a lot of files registerd """
+        if benchmark.disabled:
+            benchmark(print, "skipping, only benchmark large file counts")
+            return  # _only_ run this test is profiling
```

Comment:
```suggestion
            return  # _only_ run this test during profiling
```
Or maybe I did not understand the sentence? It's used elsewhere